pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -292,6 +321,187 @@ func (a Adapter) pushLocal(path string, files []string, delFiles []string, isFor
 	return nil
 }
 
+// Push syncs source code from the user's disk to the component
+func (a Adapter) execDevfile(pushDevfileCommands []versionsCommon.DevfileCommand, componentExists, show bool) error {
+	reader, writer := io.Pipe()
+	buildComponent := true
+	var s *log.Status
+
+	glog.V(3).Infof("mjf  execDevfile pushDevfileCommands %v", pushDevfileCommands)
+
+	for i := 0; i < len(pushDevfileCommands); i++ {
+		command := pushDevfileCommands[i]
+
+		// Exec the devBuild command if buildComponent is true
+		if (reflect.DeepEqual(command.Name, common.DefaultDevfileBuildCommand) || reflect.DeepEqual(command.Name, a.devfileBuildCmd)) && buildComponent {
+			glog.V(3).Infof("mjf executing command %v", command.Name)
+			var cmdOutput string
+
+			for _, action := range command.Actions {
+				// Change to the workdir and execute the command
+				cmdArr := []string{"/bin/sh", "-c", "cd " + *action.Workdir + " && " + *action.Command}
```

Comment:
`workdir` is an optional field in devfile
If it is not specified the command should be executed in default image directory (no `cd`ing just executing the command)

Reply:
also `action.Command` is a pointer, so it can be `nil`, so you should check it before using it, otherwise, it can end with panic

Reply:
thx, i have already done that in my new commit which i didnt push

Reply:
i shall make the changes for workdir 👍